### PR TITLE
Simplify code and fix potential NPE

### PR DESCRIPTION
**Pull request by chanseokoh**

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/internal/properties/DataflowPropertyTester.java [hunk 1]
```diff
@@ -81,12 +80,10 @@ private boolean testIsInDataflowProject(Object receiver) {
 
   private boolean isProjectDataflowProject(IProject project) {
     try {
-      if (project != null && DataflowJavaProjectNature.hasDataflowNature(project)) {
-        return true;
-      }
+      return DataflowJavaProjectNature.hasDataflowNature(project);
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/internal/properties/DataflowPropertyTester.java]: Null-check is redundant. `hasDataflowNature`'s Javadoc:

```javadoc
  /**
   * Returns true if the project is non-null, accessible (according to {@code
   * IProject#isAccessible()}), and the project description contains the Dataflow Nature ID.
   */
```

File: plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/natures/DataflowJavaProjectNature.java [hunk 2]
```diff
@@ -59,67 +57,32 @@ public void setProject(IProject project) {
    */
   public static void addDataflowJavaNatureToProject(IProject project, IProgressMonitor monitor)
       throws CoreException {
-    if (isProjectAccessible(project)) {
+    Preconditions.checkNotNull(project);
+    if (!project.isAccessible()) {
       throw new CoreException(new Status(Status.WARNING, DataflowCorePlugin.PLUGIN_ID,
-          "Can't add the Dataflow nature to nonexistent or closed project " + project.getName()));
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/natures/DataflowJavaProjectNature.java]: Previously, `project` could potentially be `null`, and a null project brings execution here. `project.getName()` would then throw an NPE.

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/handler/ModifyDataflowVersionHandler.java [hunk 3]
```diff
@@ -64,8 +57,12 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
     if (Strings.isNullOrEmpty(commandId)) {
       return null;
     }
-    ISelection selection = HandlerUtil.getCurrentSelection(event);
-    IProject project = getFirstProjectInSelection(selection);
+    IProject project = ProjectFromSelectionHelper.getFirstProject(event);
+    if (project == null) {
+      throw new ExecutionException(
+          "Exception while modifying Dataflow Maven Dependency: no project selected");
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/handler/ModifyDataflowVersionHandler.java]: This exception message comes from the code a bit below in this method. That said, it basically throws the same exception as before if `project` is `null`.

File: plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/natures/DataflowJavaProjectNature.java [hunk 4]
```diff
@@ -59,67 +57,32 @@ public void setProject(IProject project) {
...
-      return;
+          "Can't add the Dataflow nature to closed project " + project.getName()));
     }
-    IProjectDescription description = project.getDescription();
-    String[] natureIds = description.getNatureIds();
-    String[] newNatureIds = Arrays.copyOf(natureIds, natureIds.length + 1);
-    newNatureIds[natureIds.length] = DATAFLOW_NATURE_ID;
-    description.setNatureIds(newNatureIds);
-    project.setDescription(description, monitor);
+    NatureUtils.addNature(project, DATAFLOW_NATURE_ID, monitor);
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/natures/DataflowJavaProjectNature.java]: `addNature` don't add a nature if a project already has it.

**Comment by elharo** [comment]
I wonder why the kokoro build hasn't finished for 3 days?

File: plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/GpeMigrator.java [hunk 5]
```diff
@@ -123,7 +123,7 @@ private static boolean isGpeClasspath(IClasspathEntry entry) {
   @VisibleForTesting
   static boolean removeGpeNature(IProject project) throws CoreException {
     boolean hadNature = NatureUtils.hasNature(project, GPE_GAE_NATURE_ID);
-    NatureUtils.removeNature(project, GPE_GAE_NATURE_ID);
+    NatureUtils.removeNature(project, GPE_GAE_NATURE_ID, null /* monitor */);
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/GpeMigrator.java]: Looks like you could pass a monitor into removeGpeNature instead.

File: plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/internal/properties/DataflowPropertyTester.java (hunk 1 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/internal/properties/DataflowPropertyTester.java]: One of these days, we should think about using Nullable where appropriate. In fact, https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/2155

File: plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/natures/DataflowJavaProjectNature.java [hunk 6]
```diff
@@ -59,67 +57,32 @@ public void setProject(IProject project) {
    */
   public static void addDataflowJavaNatureToProject(IProject project, IProgressMonitor monitor)
       throws CoreException {
-    if (isProjectAccessible(project)) {
+    Preconditions.checkNotNull(project);
+    if (!project.isAccessible()) {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/natures/DataflowJavaProjectNature.java]: It looks like the boolean has reversed here? On the left this happens if the project is accessible, here not accessible.

Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/natures/DataflowJavaProjectNature.java]: Forgot to mention this. The wrongly named old method should have been "isProjectNotAssible()" or its test should have been reversed.

**Comment by chanseokoh** [comment]
I have observed a few times that the Kokoro status fails to be updated on GitHub due to connection errors.

BTW, I will fix up the monitor.

**Comment by chanseokoh** [comment]
Updated to check monitor cancellation. I hope this change makes sense.

**Comment by chanseokoh** [comment]
Forgot to fix tests.

**Pull request merged**